src/Symfony/Component/HttpKernel/EventListener/AbstractSessionListener.php

```diff
@@ -60,13 +60,20 @@ public function onKernelResponse(FilterResponseEvent $event)
             return;
         }
 
+        $response = $event->getResponse();
+
         if ($session->isStarted() || ($session instanceof Session && $session->hasBeenStarted())) {
-            $event->getResponse()
-                ->setPrivate()
-                ->setMaxAge(0)
-                ->headers->addCacheControlDirective('must-revalidate');
+            if (!$response->headers->has('Symfony-NoPrivateIfSessionStarted')) {
```

Comment:
This would allow cache poisoning if an external request comes with this header, isn't it?

Reply:
Why would the response contain such a header based on an external request?

Reply:
OK no sorry, I read it as a "request". What about using a request parameter instead?

Reply:
This check should also be moved upper in the method.
The name of the parameter should be more accurate, eg `session.disable_auto_cache_control`?

Reply:
I don't think this is correct. The question in that piece of code is 

> Hello response. I will now make you private if the session is started because that's what's a sane default. However, if someone wanted you to really be public even though the session is started, I'll leave you alone.

The subject is the response, not the request. Normally caching systems would add cache tagging headers to the response, set the cache control headers etc. It's all done on the response, never the request.

Reply:
> This check should also be moved upper in the method.
> The name of the parameter should be more accurate, eg session.disable_auto_cache_control or better?

Why upper? Where exactly do you mean? And the parameter can be changed yes but as it's a header I think it should be more in the header naming style, no?